collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -237,7 +237,9 @@ private void processBeanValue(
                 if (valu instanceof CompositeData) {
                     CompositeData composite = (CompositeData) valu;
                     for (String idx : rowKeys) {
-                        l2s.put(idx, composite.get(idx).toString());
+                        if ((composite != null) && (composite.get(idx) != null)) {
```

Comment:
I don't see how composite can be null here, it'd have failed the instanceof. What you want to do is call `composite.get(idx)` only once.

Reply:
You are right. I misunderstood your last comment. Fixing it now.